csbuild/_build/project.py

```diff
@@ -104,7 +104,7 @@ def __init__(self, name, workingDirectory, depends, priority, ignoreDependencyOr
 			log.Build("Preparing build tasks for {}", self)
 
 			#: type: list[Tool]
-			self.tools = projectSettings["tools"]
+			self.tools = projectSettings["tools"] - projectSettings.get("disabledTools", set())
```

Comment:
Is the type hint above this correct?  Assuming `projectSettings["tools"]` is a list, this would throw an exception if `projectSettings["disabledTools"]` doesn't exist, resulting in the default value of an empty set is returned.  If they're all sets, then `self.tools` would end up also being one.

Reply:
You're correct. It's a set. Updated.